library/core/src/main/java/com/google/android/exoplayer2/video/MediaCodecVideoRenderer.java

```diff
@@ -405,7 +405,11 @@ protected void onPositionReset(long positionUs, boolean joining) throws ExoPlayb
 
   @Override
   public boolean isReady() {
-    if (super.isReady() && (renderedFirstFrame || (dummySurface != null && surface == dummySurface)
+//        if (super.isReady() && (renderedFirstFrame || (dummySurface != null && surface == dummySurface)
+    // This "works around" the problem, even if it is not time to render a frame, and we don't have samples
+    // (because the entire LoadControl limited buffered sample queue is in the decoder), stay 'ready' to render the frame
+    // when it is due.
+    if ((super.isReady() || buffersInCodecCount > 0) && (renderedFirstFrame || (dummySurface != null && surface == dummySurface)
         || getCodec() == null || tunneling)) {
```

Comment:
The 'fix' is to keep reporting ready, even if we have fully buffered samples, as long as buffers are in the decoder that have not been presented to the output.

Reply:
The obvious drawback is the main `doSomeWork()` loop will not transition to buffering playback state in the event there are decoded buffers that have not been rendered.